diff --git a/crates/editor/src/element.rs b/crates/editor/src/element.rs
index beeed85bf0..f819b29757 100644
--- a/crates/editor/src/element.rs
+++ b/crates/editor/src/element.rs
@@ -8012,7 +8012,7 @@ pub fn render_breadcrumb_text(
     });
 
     let breadcrumbs_stack = h_flex()
-        .gap_1()
+        .gap_1p5()
         .when(multibuffer_header, |this| {
             this.pl_2()
                 .border_l_1()
diff --git a/crates/workspace/src/toolbar.rs b/crates/workspace/src/toolbar.rs
index b14775f289..afa2d702c2 100644
--- a/crates/workspace/src/toolbar.rs
+++ b/crates/workspace/src/toolbar.rs
@@ -6,7 +6,7 @@ use gpui::{
 use language::LanguageRegistry;
 use std::sync::Arc;
 use ui::prelude::*;
-use ui::{h_flex, v_flex};
+use ui::{Tab, h_flex, v_flex};
 
 pub struct PaneSearchBarCallbacks {
     pub setup_search_bar:
@@ -120,24 +120,20 @@ impl Render for Toolbar {
         v_flex()
             .group("toolbar")
             .relative()
-            .py(DynamicSpacing::Base06.rems(cx))
             .px(DynamicSpacing::Base08.rems(cx))
-            .when(has_left_items || has_right_items, |this| {
-                this.gap(DynamicSpacing::Base06.rems(cx))
-            })
             .border_b_1()
             .border_color(cx.theme().colors().border_variant)
             .bg(cx.theme().colors().toolbar_background)
             .when(has_left_items || has_right_items, |this| {
                 this.child(
                     h_flex()
-                        .items_start()
+                        .h(Tab::container_height(cx))
+                        .items_center()
                         .justify_between()
                         .gap(DynamicSpacing::Base08.rems(cx))
                         .when(has_left_items, |this| {
                             this.child(
                                 h_flex()
-                                    .min_h_8()
                                     .flex_auto()
                                     .justify_start()
                                     .overflow_x_hidden()
@@ -147,7 +143,6 @@ impl Render for Toolbar {
                         .when(has_right_items, |this| {
                             this.child(
                                 h_flex()
-                                    .h_8()
                                     .flex_row_reverse()
                                     .when(has_left_items, |this| this.flex_none())
                                     .justify_end()
